app.py

```diff
@@ -68,12 +74,119 @@
     _LOGGER.info("installation=%s", RUNTIME_CONTEXT.app_installation)
 
 
+@process_event_actions("pull_request", {"opened", "reopened", "synchronize", "edited"})
+@process_webhook_payload
+async def on_pr_open_or_sync(*, action, number, pull_request, repository, sender, installation, **kwargs):
+    """React to an opened or changed PR event.
+
+    Send a status update to GitHub via Checks API.
+    """
+    _LOGGER.info(f"on_pr_open_or_sync: working on PR {pull_request['html_url']}")
+
+    github_api = RUNTIME_CONTEXT.app_installation_client
+
+    pr_head_sha = pull_request["merge_commit_sha"]
+    if pr_head_sha is None:
+        pr_head_sha = pull_request["head"]["sha"]
+
+    repo_url = pull_request["base"]["repo"]["url"]
+
+    check_runs_base_uri = f"{repo_url}/check-runs"
+
+    resp = await github_api.post(
+        check_runs_base_uri,
+        preview_api_version="antiope",
+        data={
+            "name": CHECK_RUN_NAME,
+            "head_sha": pr_head_sha,
+            "status": "queued",
+            "started_at": f"{datetime.utcnow().isoformat()}Z",
+        },
+    )
+
+    check_runs_updates_uri = f'{check_runs_base_uri}/{resp["id"]:d}'
+
+    resp = await github_api.patch(
+        check_runs_updates_uri, preview_api_version="antiope", data={"name": CHECK_RUN_NAME, "status": "in_progress"},
+    )
+
+    _LOGGER.info(
+        f"on_pr_open_or_sync: working on PR %s: sleeping a random time between 5 and 15 seconds...",
+        pull_request["html_url"],
+    )
+
+    timeDelay = random.randrange(5, 15)
+    time.sleep(timeDelay)
+
+    await github_api.patch(
+        check_runs_updates_uri,
+        preview_api_version="antiope",
+        data={
+            "name": CHECK_RUN_NAME,
+            "status": "completed",
+            "conclusion": "neutral",
+            "completed_at": f"{datetime.utcnow().isoformat()}Z",
+            "output": {
+                "title": "Thoth's Advise",
+                "text": "Ut quis occaecat commodo incididunt aliquip aliquip occaecat sit anim irure.",
+                "summary": "This is a Developer Preview Service.\n"
+                f"Id exercitation cillum ex labore. Culpa culpa minim aute ad nulla nostrud elit"
+                f"amet. Ea velit commodo magna incididunt sint eiusmod excepteur quis. Commodo est culpa"
+                f"culpa do commodo. Lorem minim consequat exercitation culpa sint mollit minim veniam"
+                f"id Lorem fugiat tempor duis.",
+            },
+        },
+    )
+
+    _LOGGER.info(
+        f"on_pr_open_or_sync: working on PR %s: finished with `thamos advise`", pull_request["html_url"],
+    )
+
+
+# We simply extend the GitHub Event set for our use case ;)
+@process_event("thoth_thamos_advise", action="finished")
```

Comment:
Do you use webhook signatures for the GitHub App? You'd have to implement payload signing in your client too.

Reply:
Ja exactly